api/src/main/java/org/apache/iceberg/UpdatePartitionSpec.java

```diff
@@ -0,0 +1,268 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.transforms.Transform;
+
+/**
+ * API for partition spec evolution.
+ * <p>
+ * When committing, these changes will be applied to the current table metadata. Commit conflicts
+ * will not be resolved and will result in a {@link CommitFailedException}.
+ */
+public interface UpdatePartitionSpec extends PendingUpdate<PartitionSpec> {
+
+  /**
+   * Clear all partition fields in the current partition spec.
+   * <p>
+   * This will create a new partition spec without any partition field.
+   * Partition field IDs is automatically assigned and will be updated during the commit.
+   * Table schema should be obtained from the current table metadata
+   *
+   * @return this for method chaining
+   */
+  UpdatePartitionSpec clear();
+
+  /**
+   * Add a new partition field with identity transform to the partition spec.
+   * <p>
+   * The partition field id is automatically assigned and will be updated during the commit.
+   *
+   * @param sourceName the field name of the source field in the {@link PartitionSpec spec's} table schema
+   * @param targetName the name of this partition field
+   * @return this for method chaining
+   */
+  UpdatePartitionSpec addIdentityField(String sourceName, String targetName);
+
+  /**
+   * Add a new partition field with identity transform to the partition spec.
+   * <p>
+   * The partition field name is automatically assigned set.
+   * The partition field id is automatically assigned and will be updated during the commit.
+   *
+   * @param sourceName the field name of the source field in the {@link PartitionSpec spec's} table schema
+   * @return this for method chaining
+   */
+  UpdatePartitionSpec addIdentityField(String sourceName);
+
+  /**
+   * Add a new partition field with year transform to the partition spec.
+   * <p>
+   * The partition field id is automatically assigned and will be updated during the commit.
+   *
+   * @param sourceName the field name of the source field in the {@link PartitionSpec spec's} table schema
+   * @param targetName the name of this partition field
+   * @return this for method chaining
+   */
+  UpdatePartitionSpec addYearField(String sourceName, String targetName);
+
+  /**
+   * Add a new partition field with year transform to the partition spec.
+   * <p>
+   * The partition field name is automatically assigned set.
+   * The partition field id is automatically assigned and will be updated during the commit.
```

Comment:
Callers don't need to worry about field IDs, so we can remove these comments.

Reply:
👌